packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -820,7 +821,7 @@ export class GarbageCollector implements IGarbageCollector {
 				if (gcDataSuperSet.gcNodes[sourceNodeId] === undefined) {
 					gcDataSuperSet.gcNodes[sourceNodeId] = outboundRoutes;
 				} else {
-					gcDataSuperSet.gcNodes[sourceNodeId].push(...outboundRoutes);
+					gcDataSuperSet.gcNodes[sourceNodeId]?.push(...outboundRoutes);
```

Comment:
Not sure about this one - We know it's not `undefined` because of the `if` check above.  I find it misleading to add the `?`, it makes it look like `push` might not be called.

Reply:
(I'm ok with it in the tests below because they're tests)

Reply:
\Thank you for the catch, I dont think this was needed, I fixed the eslint plugin rule so this isnt needed